server/plugin/api.go

```diff
@@ -1223,6 +1291,123 @@ func (p *Plugin) handlePipelineApproveOrRejectRequest(w http.ResponseWriter, r *
+				return
+			}
+
+			if updatePipelineReleaseApprovalPostErr := p.UpdatePipelineRunApprovalPost(pipelineApprovalDetails.ApprovalSteps, pipelineApprovalDetails.MinRequiredApprovers, pipelineApprovalDetails.Status, postActionIntegrationRequest.PostId, mattermostUserID); updatePipelineReleaseApprovalPostErr != nil {
+				p.handlePipelineApprovalRequestUpdateError(constants.ErrorUpdatingPipelineApprovalRequest, mattermostUserID, updatePipelineReleaseApprovalPostErr)
+				p.handleError(w, r, &serializers.Error{Code: http.StatusInternalServerError, Message: updatePipelineReleaseApprovalPostErr.Error()})
+				return
+			}
+
+			alreadyUpdatedInformationPost := &model.Post{
+				UserId:    p.botUserID,
+				ChannelId: postActionIntegrationRequest.ChannelId,
+				Message:   "Looks like you do not have any pending approvals or have insufficient permissions for this resource.",
+			}
+			if statusCode == http.StatusConflict {
+				alreadyUpdatedInformationPost.Message = "Approval is already in completed state."
+			}
+			_ = p.API.SendEphemeralPost(mattermostUserID, alreadyUpdatedInformationPost)
+		}
+
+	default:
+		p.handlePipelineApprovalRequestUpdateError(constants.GenericErrorMessage, mattermostUserID, updatePipelineApprovalRequestErr)
+		p.handleError(w, r, &serializers.Error{Code: http.StatusInternalServerError, Message: updatePipelineApprovalRequestErr.Error()})
+		return
+	}
+
+	alphaPost.Message = "Your approval/rejection request is processed."
+	_ = p.API.UpdateEphemeralPost(mattermostUserID, alphaPost)
+	response := &model.PostActionIntegrationResponse{}
+	p.returnPostActionIntegrationResponse(w, response)
+}
+
+func (p *Plugin) UpdatePipelineRunApprovalPost(approvalSteps []*serializers.ApprovalSteps, minRequiredApprovers int, status, postID, mattermostUserID string) error {
+	post, _ := p.API.GetPost(postID)
+	slackAttachment := post.Attachments()[0]
+	noOFApprovalsReached := 0
+
+	approvers := ""
+	for _, pipelineRunApprovalSteps := range approvalSteps {
+		if pipelineRunApprovalSteps.Status != "pending" {
+			approvers += fmt.Sprintf("%s %s \n", constants.PipelineRequestUpdateEmoji[pipelineRunApprovalSteps.Status], pipelineRunApprovalSteps.AssignedApprover.DisplayName)
+			if pipelineRunApprovalSteps.Status == "approved" {
+				noOFApprovalsReached += 1
+			}
+		} else {
+			approvers += pipelineRunApprovalSteps.AssignedApprover.DisplayName + "\n"
+		}
+	}
+
+	slackAttachment.Fields = []*model.SlackAttachmentField{
+		slackAttachment.Fields[0],
+		slackAttachment.Fields[1],
+		{
+			Title: slackAttachment.Fields[2].Title,
+			Value: approvers,
+		},
+	}
+
+	if status != "pending" || noOFApprovalsReached == minRequiredApprovers {
+		slackAttachment.Actions = nil
+	}
+
+	model.ParseSlackAttachment(post, []*model.SlackAttachment{slackAttachment})
+	if _, err := p.API.UpdatePost(post); err != nil {
+		p.handlePipelineApprovalRequestUpdateError("Error in updating post", mattermostUserID, err)
```

Comment:
shouldn't there be `postID` instead of `mattermostUserID`?

Reply:
this is an util method inside this user-ID is used for DM, check that